app/src/server/api/routers/datasets.router.ts

```diff
@@ -93,16 +96,35 @@ export const datasetsRouter = createTRPCRouter({
 
       const totalTestingCount = await prisma.datasetEntry.count({
         where: {
-          datasetId: id,
+          datasetId,
           outdated: false,
           split: "TEST",
         },
       });
 
+      const { cost, inputCost, outputCost } = calculateCost(
```

Comment:
One step that I forgot to mention is calculating the total number of epochs through which we'll run our training data. If the user has only a very small amount of data, we'll actually reuse it on multiple epochs.

If you search the codebase for `calculateNumEpochs`, you can see where we're already doing this in checkFineTuneStatus.task.ts.